v22.2/trigram-indexes.md

```diff
@@ -0,0 +1,354 @@
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE INDEX ON t USING GIN (w gin_trgm_ops);
+~~~
+
+Check that the statement uses the trigram index:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+EXPLAIN SELECT w, similarity(w, 'word')
+  FROM t
+  WHERE w % 'word'
+  ORDER BY similarity DESC, w;
+~~~
+
+~~~
+                                                                         info
+-------------------------------------------------------------------------------------------------------------------------------------------------------
+  distribution: local
+  vectorized: true
+
+  • sort
+  │ estimated row count: 3,337
+  │ order: -similarity,+w
+  │
+  └── • render
+      │
+      └── • filter
+          │ estimated row count: 3,337
+          │ filter: w % 'word'
+          │
+          └── • index join
+              │ estimated row count: 6
+              │ table: t@t_pkey
+              │
+              └── • inverted filter
+                  │ estimated row count: 6
+                  │ inverted column: w_inverted_key
+                  │ num spans: 2
+                  │
+                  └── • scan
+                        estimated row count: 6 (0.06% of the table; stats collected 8 minutes ago; using stats forecast for 24 minutes in the future)
+                        table: t@t_w_idx
+                        spans: 2 spans
+~~~
+
+Execute the statement again and note the improved performance:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT w, similarity(w, 'word')
+  FROM t
+  WHERE w % 'word'
+  ORDER BY similarity DESC, w;
+~~~
+
+~~~
+     w    |     similarity
+----------+----------------------
+  wordy   |  0.5714285714285714
+  wofoord |  0.4444444444444444
+  worm    | 0.42857142857142855
+  wort    | 0.42857142857142855
+  world   |               0.375
+  worried |                 0.3
+(6 rows)
+
+
+Time: 4ms total (execution 4ms / network 0ms)
+~~~
+
+Pattern matching with `LIKE` and `ILIKE` is also accelerated by a trigram index:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+EXPLAIN SELECT * FROM t WHERE w LIKE '%foo%';
+~~~
+
+~~~
+                                                                    info
+---------------------------------------------------------------------------------------------------------------------------------------------
+  distribution: local
+  vectorized: true
+
+  • filter
+  │ estimated row count: 3,337
+  │ filter: w LIKE '%foo%'
+  │
+  └── • index join
+      │ estimated row count: 0
+      │ table: t@t_pkey
+      │
+      └── • scan
+            estimated row count: 0 (<0.01% of the table; stats collected 15 minutes ago; using stats forecast for 42 minutes in the future)
+            table: t@t_w_idx
+            spans: 1 span
+~~~
+
+## Unsupported features
+
+The following PostgreSQL syntax and features are currently unsupported. For details, see the [tracking issue](https://github.com/cockroachdb/cockroach/issues/41285).
```

Comment:
Does this need to be added to the known limitations page?

Reply:
Adding per @lnhsingh